src/java/picard/sam/markduplicates/MarkDuplicates.java

```diff
@@ -578,11 +626,26 @@ private void generateDuplicateIndexes(final boolean useBarcodes, final boolean i
             maxInMemory /= 2;
             this.opticalDuplicateIndexes = new SortingLongCollection(maxInMemory, TMP_DIR.toArray(new File[TMP_DIR.size()]));
         }
+        // If we're are tracking representative reads, adjust maInMemory
+        if (TAG_DUPLICATE_SET_MEMBERS) {
+            // Memory requirements for RepresentativeReadIndexer:
+            // three int entries + overhead: (3 * 4) + 4
+            maxInMemory = (int) (Runtime.getRuntime().maxMemory() * 0.25) / 16;
+        }
         log.info("Will retain up to " + maxInMemory + " duplicate indices before spilling to disk.");
         this.duplicateIndexes = new SortingLongCollection(maxInMemory, TMP_DIR.toArray(new File[TMP_DIR.size()]));
+        if (TAG_DUPLICATE_SET_MEMBERS){
+            final RepresentativeReadIndexerCodec representativeIndexCodec = new RepresentativeReadIndexerCodec();
+            this.representativeReadIndicesForDuplicates = SortingCollection.newInstance(RepresentativeReadIndexer.class,
+                    representativeIndexCodec,
+                    new RepresentativeReadComparator(),
+                    maxInMemory,
+                    TMP_DIR);
+        }
 
         ReadEndsForMarkDuplicates firstOfNextChunk = null;
-        final List<ReadEndsForMarkDuplicates> nextChunk = new ArrayList<ReadEndsForMarkDuplicates>(200);
+        final List nextChunk;
+        nextChunk = new ArrayList<ReadEndsForMarkDuplicates>(200);
```

Comment:
I don't know the motivation for this particular value. It's a good question, but I'm not aiming to change the original behavior with this feature I'm working on.